pkg/assembler/helpers/namespaces.resolver.go

```diff
@@ -0,0 +1,31 @@
+package helpers
+
+import (
+	"fmt"
+
+	"github.com/guacsec/guac/pkg/assembler/graphql/model"
+)
+
+func UpdatePurlForNamespaces(packageObj *model.Package) ([]*model.PackageNamespace, error) {
+	updatedNamespaces := make([]*model.PackageNamespace, len(packageObj.Namespaces))
+
+	for i, namespace := range packageObj.Namespaces {
+		modifiedNamespace := &model.PackageNamespace{
+			ID:        namespace.ID,
+			Namespace: namespace.Namespace,
+		}
+		names := make([]*model.PackageName, len(namespace.Names))
+		for j, name := range namespace.Names {
+			names[j] = name
+			versions := make([]*model.PackageVersion, len(name.Versions))
+			for k, version := range name.Versions {
+				versions[k] = version
+				version.Purl = fmt.Sprintf("pkg:%s/%s/%s", packageObj.Type, namespace.Namespace, name.Name)
```

Comment:
pkg/assembler/helpers/purl.go will help with reconstructing the purl with the version, subpath and qualifiers.

Reply:
@pxp928 updated the code to use `PkgToPurl` function to generate purl.
Why the `purl` format is different than `purl` present in `sbom` file? (just curious)

Reply:
It matches. My bad, I got confused between `cpe` and `purl` field. In the `cpe` field, the format is `<name>:<version>`, while in the purl field, it is `<name>@<version>`.
Before using the `PkgToPurl` function, I was constructing the purl incorrectly by using `:` instead of `@`. Thanks for recommending the `PkgToPurl` function.